src/lib.rs

```diff
@@ -71,12 +71,22 @@ impl<R: Runtime> AppHandleExt for tauri::AppHandle<R> {
 pub trait WindowExt {
   fn restore_state(&self, auto_show: bool) -> tauri::Result<()>;
 }
+trait MonitorScale {
+  fn scale(&self, factor: f64) -> u32;
+}
+
+impl MonitorScale for u32 {
+  fn scale(&self, factor: f64) -> u32 {
+    ((*self as f64) * factor) as u32
+  }
+}
 
 impl<R: Runtime> WindowExt for Window<R> {
   fn restore_state(&self, auto_show: bool) -> tauri::Result<()> {
     let cache = self.state::<WindowStateCache>();
     let mut c = cache.0.lock().unwrap();
     let mut should_show = true;
+    let scale_factor = self.current_monitor()?.unwrap().scale_factor();
```

Comment:
you need to use the scale factor of the target monitor not the current monitor, which you can get it along with position in Line 93